index.coffee

```diff
@@ -8,8 +8,7 @@
       (q, http) ->
        req = (opts) ->
          d = q.defer()
-         http(opts).then (res) ->
-           d.resolve res.data, res
+         http(opts).then ((res) -> d.resolve res.data, res), ((res) -> d.reject res.data, res)
```

Comment:
The original idea was to have the exactly same signature for reject as the underlying implementation. To instead of ((res) -> d.reject res.data, res) we should have simply d.reject
Then the client code would handle a hybind reject it as it would handle any other $http reject. 
If ok, can you change that on your PR and I'll merge?